pai-management/k8sPaiLibrary/maintainlib/deploy.py

```diff
@@ -203,6 +203,12 @@ def run(self):
         kubectl_install_instance = kubectl_install.kubectl_install(self.cluster_config)
         kubectl_install_instance.run()
 
+        # check the registerd api resources
+        common.execute_shell_retry("kubectl api-resources", "kubectl command failed!", 5)
+
+        # create kube-proxy until daemonset resource is registerd
+        common.execute_shell_retry("kubectl api-resources | grep -q daemonsets", "Controller manager hasn't create daemonset object!", 5)
```

Comment:
1. Will `kubectl api-resources | grep -q daemonsets`  return none zero when daemonsets not ready?
2. Does the grep -q case-sensitive?

Reply:
yes, only when it grep nothing it return 0, I have test

Reply:
kubectl api-resources results are all lowercase, the "grep" is case-sensitive